src/Microsoft.Data.Sqlite.Core/SqliteParameter.cs

```diff
@@ -208,137 +204,11 @@ internal bool Bind(sqlite3_stmt stmt)
-                }
-                else if (type == typeof(double))
-                {
-                    var value = (double)_value;
-                    _bindAction = (s, i) => BindDouble(s, i, value);
-                }
-                else if (type == typeof(float))
-                {
-                    var value = (double)(float)_value;
-                    _bindAction = (s, i) => BindDouble(s, i, value);
-                }
-                else if (type == typeof(Guid))
-                {
-                    var value = ((Guid)_value).ToByteArray();
-                    _bindAction = (s, i) => BindBlob(s, i, value);
-                }
-                else if (type == typeof(int))
-                {
-                    var value = (long)(int)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(long))
-                {
-                    var value = (long)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(sbyte))
-                {
-                    var value = (long)(sbyte)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(short))
-                {
-                    var value = (long)(short)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(string))
-                {
-                    var value = (string)_value;
-                    _bindAction = (s, i) => BindText(s, i, value);
-                }
-                else if (type == typeof(TimeSpan))
-                {
-                    var value = ((TimeSpan)_value).ToString("c");
-                    _bindAction = (s, i) => BindText(s, i, value);
-                }
-                else if (type == typeof(uint))
-                {
-                    var value = (long)(uint)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(ulong))
-                {
-                    var value = (long)(ulong)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else if (type == typeof(ushort))
-                {
-                    var value = (long)(ushort)_value;
-                    _bindAction = (s, i) => raw.sqlite3_bind_int64(s, i, value);
-                }
-                else
-                {
-                    throw new InvalidOperationException(Resources.UnknownDataType(type));
-                }
-
-                _bindActionValid = true;
-            }
-
-            _bindAction(stmt, index);
+            new SqliteParameterBinder(stmt, index, _value).Bind();
```

Comment:
Why don't you store the binding action anymore? This will slow down the reuse of prepared statements,

Reply:
Over-optimization. We haven't done any performance profiling on the library, so we don't actually know what difference it makes. It would be nice to do an actual round of performance tuning this release.